packages/core/src/discovery.js

```diff
@@ -88,21 +88,42 @@ function waitForDiscoveryNetworkIdle(page, options) {
 
 // Creates an initial resource map for a snapshot containing serialized DOM
 function parseDomResources({ url, domSnapshot }) {
-  if (!domSnapshot) return new Map();
-  let isHTML = typeof domSnapshot === 'string';
-  let { html, resources = [] } = isHTML ? { html: domSnapshot } : domSnapshot;
-  let rootResource = createRootResource(url, html);
+  const map = new Map();
+  if (!domSnapshot) return map;
+  let allResources = new Set();
+
+  if (Array.isArray(domSnapshot)) {
+    let allRootResources = new Set();
+    for (let snapshot of domSnapshot) {
+      const dom = snapshot.domSnapshot;
+      const width = snapshot.width;
+      let isHTML = typeof dom === 'string';
+      let { html, resources = [] } = isHTML ? { html: dom } : dom;
+      resources.forEach(r => allResources.add(r));
+      let rootResource = createRootResource(url, html, { widths: [width] });
+      allRootResources.add(rootResource);
+    }
+    allRootResources = Array.from(allRootResources);
+    map.set(allRootResources[0].url, allRootResources);
+    allResources = Array.from(allResources);
+  } else {
+    let isHTML = typeof domSnapshot === 'string';
+    let { html, resources = [] } = isHTML ? { html: domSnapshot } : domSnapshot;
+    allResources = resources;
+    let rootResource = createRootResource(url, html);
+    map.set(rootResource.url, rootResource);
+  }
 
   // reduce the array of resources into a keyed map
-  return resources.reduce((map, { url, content, mimetype }) => {
+  return allResources.reduce((map, { url, content, mimetype }) => {
```

Comment:
this is accurate ? can we have same resource url across multiple widths ?

Reply:
So instead of rootResource if we capture any other resource with a url. we assume it will have same content. 
this is same as current behaviour of deferUploads where we don't override a url if it's already present in resources map. refer: https://github.com/percy/cli/blob/676420ece5f33bfa1e3d8be223abdd8093facb77/packages/core/src/snapshot.js#L303
Even if we start storing resources for different widths our rendering can only pick one resource for a given url.

Reply:
What about the custom resources like CSSOM that we gather ? do we uniquely name it ? and would correct width get added to correct root resource ?

Reply:
Checked for all captured resources(in serializeDOM). we always add a uid (random 36 character string). so it should always be unique url.
serialized canvas - uid (data-percy-element-id)
serialized videos - uid (data-percy-element-id)
stylesheets - uid